src/Popover.js

```diff
@@ -1,79 +1,177 @@
+  toggle: () => {},
 };
 
 class Popover extends React.Component {
   constructor(props) {
     super(props);
 
-    this.getTetherConfig = this.getTetherConfig.bind(this);
+    this.addTargetEvents = this.addTargetEvents.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+    this.removeTargetEvents = this.removeTargetEvents.bind(this);
+    this.toggle = this.toggle.bind(this);
+    this.show = this.show.bind(this);
+    this.hide = this.hide.bind(this);
+  }
+
+  componentDidMount() {
+    this._target = getTarget(this.props.target);
+    this.handleProps();
+  }
+
+  componentDidUpdate() {
+    this.handleProps();
+  }
+
+  componentWillUnmount() {
+    this.clearShowTimeout();
+    this.clearHideTimeout();
+    this.removeTargetEvents();
+  }
+
+  getDelay(key) {
+    const { delay } = this.props;
+    if (typeof delay === 'object') {
+      return isNaN(delay[key]) ? DEFAULT_DELAYS[key] : delay[key];
+    }
+    return delay;
   }
 
-  getTetherConfig() {
-    const attachments = getTetherAttachments(this.props.placement);
-    return {
-      ...defaultTetherConfig,
-      ...attachments,
-      target: '#' + this.props.target,
-      ...this.props.tether
-    };
+  handleProps() {
+    if (this.props.isOpen) {
+      this.show();
+    } else {
+      this.hide();
+    }
+  }
+
+  show() {
+    this.clearHideTimeout();
+    this.addTargetEvents();
+    if (!this.props.isOpen) {
+      this.clearShowTimeout();
+      this._showTimeout = setTimeout(this.toggle, this.getDelay('show'));
+    }
+  }
+
+  hide() {
+    this.clearShowTimeout();
+    this.removeTargetEvents();
+    if (this.props.isOpen) {
+      this.clearHideTimeout();
+      this._hideTimeout = setTimeout(this.toggle, this.getDelay('hide'));
+    }
+  }
+
+  clearShowTimeout() {
+    clearTimeout(this._showTimeout);
+    this._showTimeout = undefined;
+  }
+
+  clearHideTimeout() {
+    clearTimeout(this._hideTimeout);
+    this._hideTimeout = undefined;
+  }
+
+  handleDocumentClick(e) {
+    if (e.target !== this._target && !this._target.contains(e.target)) {
+      if (this._hideTimeout) {
+        this.clearHideTimeout();
+      }
+
+      if (this.props.isOpen) {
+        this.toggle();
+      }
+    }
+  }
+
+  addTargetEvents() {
+    ['click', 'touchstart'].forEach(event =>
+      document.addEventListener(event, this.handleDocumentClick, true)
+    );
+  }
+
+  removeTargetEvents() {
+    ['click', 'touchstart'].forEach(event =>
+      document.removeEventListener(event, this.handleDocumentClick, true)
+    );
+  }
+
+  toggle(e) {
+    if (this.props.disabled) {
+      return e && e.preventDefault();
+    }
+
+    return this.props.toggle();
   }
 
   render() {
     if (!this.props.isOpen) {
       return null;
     }
 
-    let tetherConfig = this.getTetherConfig();
-
+    const attributes = omit(this.props, Object.keys(propTypes));
     const classes = mapToCssModules(classNames(
       'popover-inner',
       this.props.className
```

Comment:
Shouldn't `this.props.className` be on the popperClasses instead?
Currently I get the following scenario:

```jsx
<Popover
    className="day-picker-popover"
    placement="bottom"
    isOpen={open}
>
```

Outputs:
```html
<div data-reactroot="" class="popover show bs-popover-bottom" style="position: absolute; transform: translate3d(176px, 489px, 0px); top: 0px; left: 0px; will-change: transform;" data-placement="bottom">
    <div class="popover-inner day-picker-popover">
```

I expected "day-picker-popover" to be on the parent element not the inner as I can't use the classname to. override popover styles in this case.

Reply:
Yeah, it probably makes more sense to have that go apply to the popper and create a new prop to extend the inner popover classes.